deps/rabbit/src/rabbit_stream_coordinator.erl

```diff
@@ -1273,7 +1408,56 @@ eval_listeners(#stream{listeners = Listeners0,
             {Stream#stream{listeners = Listeners}, Effects};
         _ ->
             {Stream, Effects0}
-    end.
+    end;
+eval_listeners(MachineVersion, #stream{listeners = Listeners0,
+                                       queue_ref = QRef,
+                                       members = Members} = Stream0, Effects0) when MachineVersion >= 2 ->
+    %% Iterating over stream listeners.
+    %% Returning the new map of listeners and the effects (notification of changes)
+    {Listeners1, Effects1} =
+        maps:fold(fun({P, leader}, ListLPid0, {Lsts0, Effs0}) ->
+                          %% iterating over member to find the leader
```

Comment:
does `find_leader/1` not do the job here?

Reply:
`find_leader/1` does not account for leader pid changes.